### PR TITLE
Add copy button to named language code blocks

### PR DESCRIPTION
**Pull request by dit7ya**
Once again, there were quite a few hoops to pass through.

With the `jsx: true` option in the transformer, it disables the onclick handler on the transformer, which we manually add later on each page that contains the correct class name. See the implementation of the `registerCopyButton` function to understand what's going on here.

I expected docusaurus component swizzling to work, but adding the registerCopyButton as a hook (as the `@rehype-pretty/transformers` library suggests) to /theme/Root.tsx or /theme/Layout/index.tsx didn't do the job.

Note: The style of the copy-button is different for named codeblocks, and I couldn't find any easy way to match the styles completely (even though https://github.com/rehype-pretty/rehype-pretty-code/blob/eee1b216347fe4c1181ddf80bfa6a6a8e427ffdd/packages/transformers/src/copy-button.ts#L15 exists, the style for the borders is not controllable). You could simply use the language name `txt` for plaintext codeblocks moving forward.

### CODE REVIEW

**Comment by dit7ya** [comment]
I just noticed there are a few styling issues, I will fix them tomorrow.

**Comment by mavam** [comment]
> I just noticed there are a few styling issues, I will fix them tomorrow.

I've haven't seen activity on this branch since that last comment. Can you please push this over the finish line?

**Comment by dit7ya** [comment]
This PR is now ready for review.

**Comment by mavam** [comment]
Looks like this to me:

image

Close up:

image

I don't find the design compelling, TBH. The black boundary doesn't look good. Any reason we are not using the typical icon for this, like the one GitHub is using?

image

File: web/registerCopyButton.js
Comment (dominiklohmann) [inline web/registerCopyButton.js:1]: The design for this looks very off to me:

image

**Review by mavam — APPROVED** [review]
Tried on Chrome and Safari, works!

File: web/registerCopyButton.js
Comment (mavam) [reply web/registerCopyButton.js:1]: Just tested on Safari after @dit7ya fixup. This is no longer an issue.

Comment (dit7ya) [reply web/registerCopyButton.js:1]: Matthias says it works on Safari.

**Pull request merged**